docs/fsharp/language-reference/plaintext-formatting.md

```diff
@@ -0,0 +1,434 @@
+output themselves contain linebreaks.  For example
+
+```fsharp
+printfn "%0A" [| for i in 1 .. 5 -> (i, i*i) |]
+
+printfn "%0A" [| for i in 1 .. 5 -> "abc\ndef |]
+```
+
+produces
+
+```console
+[|(1, 1); (2, 4); (3, 9); (4, 16); (5, 25)|]
+[|"abc
+def"; "abc
+def"; "abc
+def"; "abc
+def"; "abc
+def"|]
+```
+
+A depth limit of 4 is used for sequence (`IEnumerable`) values, which are shown as `seq { ...}`. A depth limit of 100 is used for list and array values.
+For example,
+
+```fsharp
+printfn "%A" (seq { for i in 1 .. 10 -> (i, i*i) })
+```
+
+produces
+
+```console
+seq [(1, 1); (2, 4); (3, 9); (4, 16); ...]
+```
+
+Block indentation is also used for the the structure of public record and union values. For example,
+
+```fsharp
+type R = { X : int list; Y : string list }
+
+printfn "%A" { X =  [ 1;2;3 ]; Y = ["one"; "two"; "three"] }
+```
+
+produces
+
+```console
+{ X = [1; 2; 3]
+  Y = ["one"; "two"; "three"] }
+```
+
+If `%+A` is used, then the private structure of records and unions
+is also revealed by using reflection. For example
+
+```fsharp
+type internal R =
+    { X : int list; Y : string list }
+    override x.ToString() = "R"
+
+let internal data = { X = [ 1;2;3 ]; Y = ["one"; "two"; "three"] }
+
+printfn "external view:\n%A" data
+
+printfn "internal view:\n%+A" data
+
+```
+
+produces
+
+```console
+external view:
+R
+
+internal view:
+{ X = [1; 2; 3]
+  Y = ["one"; "two"; "three"] }
+```
+
+### `%A` formatting of large, cyclic, or deeply nested values
+
+Large structured values are formatted to a maximum overall object node count of 10000.
+Deeply nested values are formatted to a depth of 100.  In both cases `...` is used
+to elide some of the output.  For example,
+
+```fsharp
+type Tree = Node of Tree * Tree | Tip
+
+let rec make n = if n = 0 then Tip else Node(Tip, make (n-1))
+
+printfn "%A" (make 1000)
+```
+
+produces a large output with some parts elided:
+
+```console
+Node(Tip, Node(Tip, ....Node (..., ...)...))
+```
+
+Cycles are detected in the object graphs and `...` is used at places where cycles are detected. For example
+
+```fsharp
+type R = { mutable Links: R list }
+let r = { Links = [] }
+r.Links <- [r]
+printfn "%A" r
+```
+
+produces
+
+```console
+{ Links = [...] }
+```
+
+### `%A` formatting of lazy, null, and function values
+
+Lazy values are printed as `Value is not created` or equivalent text when the value has not yet been evaluated.
+
+Null values are printed as `null` unless the static type of the value is determined to be a union type where `null` is
+a permitted represenation.
+
+F# function values are printed as their internally generated closure name, for example, `<fun:it@43-7>`.
+
+### Customize plain text formatting with `StructuredFormatDisplayAttribute`
+
+When using the `%A` specifier, the presence of `StructuredFormatDisplayAttribute` on type declarations is respected.  This can
```

Comment:
```suggestion
When using the `%A` specifier, the presence of the `StructuredFormatDisplay` attribute on type declarations is respected.  This can
```